src/EditorFeatures/Core/DocumentationComments/AbstractDocumentationCommentCommandHandler.cs

```diff
@@ -97,13 +125,292 @@ private bool CompleteComment(
+                {
+                    ThreadingContext.ThrowIfNotOnUIThread();
+
+                    var token = _asyncListener.BeginAsyncOperation(nameof(GenerateDocumentationProposalAsync));
+                    _ = GenerateDocumentationProposalAsync(document, snippet, oldSnapshot, oldCaret, cancellationToken).CompletesAsyncOperation(token);
+                }
             }
         }
 
         return returnValue;
     }
 
+    private async Task GenerateDocumentationProposalAsync(Document document, DocumentationCommentSnippet snippet,
+        ITextSnapshot oldSnapshot, VirtualSnapshotPoint oldCaret, CancellationToken cancellationToken)
+    {
+        await Task.Yield().ConfigureAwait(false);
+
+        // Bailing out if copilot is not available or the option is not enabled.
+        if (document.GetRequiredLanguageService<ICopilotCodeAnalysisService>() is not { } copilotService ||
+                await copilotService.IsAvailableAsync(cancellationToken).ConfigureAwait(false) is false)
+        {
+            return;
+        }
+
+        if (document.GetLanguageService<ICopilotOptionsService>() is not { } copilotOptionService ||
+            !await copilotOptionService.IsGenerateDocumentationCommentOptionEnabledAsync().ConfigureAwait(false))
+        {
+            return;
+        }
+
+        var snippetProposal = GetSnippetProposal(snippet.SnippetText, snippet.MemberNode, snippet.Position, snippet.CaretOffset);
+
+        if (snippetProposal is null)
+        {
+            return;
+        }
+
+        // Do not do IntelliCode line completions if we're about to generate a documentation comment
+        // so that won't have interfering grey text.
+        _intellicodeLineCompletionsDisposable = await _suggestionManagerBase!.DisableProviderAsync(SuggestionServiceNames.IntelliCodeLineCompletions, cancellationToken).ConfigureAwait(false);
+
+        var proposalEdits = await GetProposedEditsAsync(snippetProposal, copilotService, oldSnapshot, snippet.IndentText, cancellationToken).ConfigureAwait(false);
+
+        var proposal = Proposal.TryCreateProposal(null, proposalEdits, oldCaret, flags: ProposalFlags.SingleTabToAccept);
+
+        if (proposal is null)
+        {
+            return;
+        }
+
+        var suggestion = new DocumentationCommentSuggestion(this, proposal);
+
+        var session = this._suggestionSession = await (_suggestionManagerBase.TryDisplaySuggestionAsync(suggestion, cancellationToken)).ConfigureAwait(false);
+
+        if (session != null)
+        {
+            await TryDisplaySuggestionAsync(session, suggestion, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    /// <summary>
+    /// Traverses the documentation comment shell and retrieves the pieces that are needed to generate the documentation comment.
+    /// </summary>
+    private static DocumentationCommentProposal? GetSnippetProposal(string? comments, SyntaxNode? memberNode, int? position, int caret)
```

Comment:
It seems this entire method is just parsing the stub comment generated by `AbstractDocumentationCommentSnippetService.GetDocumentationCommentStubLines`. Could we let it create the `DocumentationCommentProposedEdit`s directly?